src/unix/fsevents.c

```diff
@@ -267,12 +271,6 @@ static void uv__fsevents_event_cb(ConstFSEventStreamRef streamRef,
         }
       }
 
-#ifdef MAC_OS_X_VERSION_10_7
-      /* Ignore events with path equal to directory itself */
-      if (len == 0)
```

Comment:
Oh disregard my comment about files, I see why it is necessary to remove this condition for them.

It will emit an extra event for all directories, whenever a new file is created or removed in them. It seems to me that we are not really interested in this event. Perhaps we could check if the `handle` represents file here?